plugin/cog/cog-rangereader-azure/src/main/java/it/geosolutions/imageioimpl/plugins/cog/AzureConfigurationProperties.java

```diff
@@ -0,0 +1,156 @@
+import java.net.URI;
+import java.net.UnknownHostException;
+
+/**
+ * Helps locate configuration properties in system/environment for use in building Azure client.
+ */
+public class AzureConfigurationProperties {
+
+    private static final String AZURE_ACCOUNT_NAME = "azure.reader.accountName";
+    private static final String AZURE_ACCOUNT_KEY = "azure.reader.accountKey";
+    private static final String AZURE_ACCOUNT_CONTAINER = "azure.reader.container";
+    private static final String AZURE_ACCOUNT_PREFIX = "azure.reader.prefix";
+    private static final String AZURE_MAX_CONNECTIONS = "azure.reader.maxConnections";
+
+    private String container;
+    private String prefix;
+    private String accountName;
+    private String accountKey;
+    private Integer maxConnections = 5;
+    private Boolean useHTTPS = Boolean.TRUE;
+    private String serviceURL = null;
+
+    AzureConfigurationProperties(BasicAuthURI cogUri) {
+        URI uri = cogUri.getUri();
+
+        // sample uri
+        // https://myaccount.blob.core.windows.net/mycontainer/prefix/myblob.tif
+        String path = uri.toASCIIString();
+
+        if (path.startsWith("https")) {
+            // replace for quicker finding
+            path = path.replace("https", "http");
+            useHTTPS = true;
+        }
+
+        BlobURLParts parts = null;
+        try {
+            parts = URLParser.parse(uri.toURL());
+            container = parts.containerName();
+            String blobName = parts.blobName();
+            int lastIndex = blobName.lastIndexOf("/");
+            if (lastIndex > 0) {
+                prefix = blobName.substring(0, lastIndex);
+            }
+            String host = parts.host();
+            int blobcoreIdx = host.indexOf("." + AzureClient.AZURE_URL_BASE);
+            if (blobcoreIdx > 0) {
+                accountName = host.substring(0, blobcoreIdx);
+            }
+
+        } catch (UnknownHostException| MalformedURLException e) {
+            throw new RuntimeException("Unable to parse the provided uri " + path + "due to " + e.getLocalizedMessage());
+        }
+
+        if (container == null) {
+            container = PropertyLocator.getEnvironmentValue(AZURE_ACCOUNT_CONTAINER, null);
+        }
+        if (prefix == null) {
+            prefix = PropertyLocator.getEnvironmentValue(AZURE_ACCOUNT_PREFIX, null);
+        }
+        if (cogUri.getUser() != null && cogUri.getPassword()!= null) {
+            accountName = cogUri.getUser();
+            accountKey = cogUri.getPassword();
+        }
+
+        if (accountName == null) {
+            accountName = PropertyLocator.getEnvironmentValue(AZURE_ACCOUNT_NAME, null);
+        }
+        if (accountKey == null) {
+            accountKey = PropertyLocator.getEnvironmentValue(AZURE_ACCOUNT_KEY, null);
+        }
+        if (maxConnections == null) {
+            maxConnections = Integer.parseInt(PropertyLocator.getEnvironmentValue(AZURE_MAX_CONNECTIONS, "5"));
```

Comment:
Normally the default max connections for browsers is 6... where does 5 come from? :rofl:

Reply:
Good question. I think that I have copied that part from a previous reference implementation. 
I'm going to default it to 10 :)